src/async_std.rs

```diff
@@ -279,7 +279,7 @@ pub fn future_into_py_with_locals<F, T>(
 ) -> PyResult<Bound<PyAny>>
 where
     F: Future<Output = PyResult<T>> + Send + 'static,
-    T: IntoPy<PyObject>,
+    T: for<'py> IntoPyObject<'py>,
```

Comment:
These bounds are too strict. You should introduce a `'py` lifetime on function and tie it to the `Python<'py>` token, as well as `IntoPyObject<'py>`. You only need higher ranked bounds of there is no `'py` input lifetime. (Same for a lot of the others)